File: Baekjoon/4단계 1차원 배열/바구니 뒤집기.py
n, m =map(int, input().split())
arr = list(range(1, n + 1))
for _ in range(m):
    i, j = map(int, input().split())
    temp = arr[i - 1:j]
    temp.reverse()
    arr[i - 1:j] = temp
print(' '.join(map(str, arr)))

File: Baekjoon/4단계 1차원 배열/바구니 뒤집기.js
const fs = require('fs');

const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const [basketCounts, changeCounts] = input[0].split(' ').map(Number);
const ballNumbers = [];
for (let i = 0; i < basketCounts; i += 1) {
  ballNumbers[i] = i + 1;
}

let temp = [];
for (let i = 1; i <= changeCounts; i += 1) {
  const [basket1, basket2] = input[i].split(' ').map(Number);
  for (let j = basket1 - 1; j < basket2; j += 1) {
    temp.push(ballNumbers[j]);
  }
  temp.reverse();
  ballNumbers.splice(basket1 - 1, basket2 - basket1 + 1, ...temp);
  temp = [];
}

console.log(ballNumbers.join(' '));
